lhotse/dataset/sampling.py

```diff
@@ -67,8 +84,13 @@ def __init__(
         self.shuffle = shuffle
         self.seed = seed
         self.epoch = 0
+
         self._maybe_init_distributed(world_size=world_size, rank=rank)
-        self.data_source = partition_cut_ids(self.full_data_source, world_size=self.world_size, rank=self.rank)
+        self.data_source = DataSource(
+            partition_cut_ids(self.full_data_source, world_size=self.world_size, rank=self.rank)
+        )
+
+        self._len = None
```

Comment:
I suggest renaming this to `num_batches` everywhere, it seems more intuitive to set in the sub-classes (like `BucketingSampler`)

Reply:
renaming